core/embed/rust/src/ui/model_tt/screens.rs

```diff
@@ -0,0 +1,107 @@
+#[cfg(feature = "micropython")]
+use crate::micropython::buffer::StrBuffer;
+use crate::ui::{
+    component::{
+        text::paragraphs::{Paragraph, ParagraphVecShort, Paragraphs, VecExt},
+        Component,
+    },
+    display::Icon,
+    geometry::LinearPlacement,
+    model_tt::{
+        component::ResultScreen,
+        constant,
+        theme::{FATAL_ERROR_COLOR, ICON_WARN_SMALL, TEXT_ERROR_BOLD, TEXT_ERROR_NORMAL, WHITE},
+    },
+    util::from_c_str,
+};
+
+#[cfg(not(feature = "micropython"))]
+fn get_str(text: &str) -> &str {
+    text
+}
+#[cfg(feature = "micropython")]
+fn get_str(text: &'static str) -> StrBuffer {
+    text.into()
+}
+
+#[no_mangle]
+extern "C" fn screen_fatal_error(msg: *const cty::c_char, file: *const cty::c_char) -> u32 {
+    let m_top = if msg.is_null() {
+        let mut messages = ParagraphVecShort::new();
+
+        messages.add(Paragraph::new(&TEXT_ERROR_BOLD, get_str("FATAL ERROR!")).centered());
+        messages.add(
+            Paragraph::new(
+                &TEXT_ERROR_NORMAL,
+                get_str(unwrap!(unsafe { from_c_str(file) })),
+            )
+            .centered(),
+        );
+
+        Paragraphs::new(messages).with_placement(LinearPlacement::vertical().align_at_center())
+    } else {
+        let msg = unwrap!(unsafe { from_c_str(msg) });
+        let mut messages = ParagraphVecShort::new();
+
+        messages.add(Paragraph::new(&TEXT_ERROR_BOLD, get_str("FATAL ERROR!")).centered());
+        messages.add(Paragraph::new(&TEXT_ERROR_NORMAL, get_str(msg)).centered());
+
+        Paragraphs::new(messages).with_placement(LinearPlacement::vertical().align_at_center())
+    };
+    let mut messages = ParagraphVecShort::new();
+
+    messages.add(
+        Paragraph::new(&TEXT_ERROR_BOLD, get_str("PLEASE CONTACT\nTREZOR SUPPORT")).centered(),
```

Comment:
Hm but then compiler can infer the type it needs to convert `Into`, so i would need some #cfg anyway

Reply:
ISTM we could declare that the error screens always accept `&str` and never `StrBuffer`, and resolve this by `let mut frame = ResultScreen::<&str>::new(...)` in the lines below.
then we don't even need the `into()`

that is nonproblematic because even if incoming data is in a `StrBuffer`, the error screens are render and forget so I can borrow the &str out of the `StrBuffer` and the borrow will end as soon as the error screen call is done

Reply:
`T` needs to be `ParagraphStrType`, and this is not implemented for `&str`outside of bootloader, as there are some flash size consequences. We might get rid of the `Paragraphs` in `ResultScreen` as suggested below and then it hopefully will be possible to simplify here too.

But i don't wanna do this right now, as there will be redesign of these screens anyway so i don't want to redo this twice.